types/public/Interfaces.d.ts

```diff
@@ -1,62 +1,162 @@
+     */
     bindings: {
         [key: string]: any;
     };
+    /**
+     * Trigger metadata and function invocation data.
+     */
     bindingData: {
         [key: string]: any;
     };
-    bindingDefinitions: {
-        [key: string]: any;
-    };
+    /**
+     * Bindings your function uses, as defined in function.json.
+     */
+    bindingDefinitions: IBindingDefinition[];
+    /**
+     * Calling directly allows you to write streaming function logs at the default trace level.
+     */
     log: ILogger;
-    req?: IRequest;
-    res?: IResponse;
+    /**
+     * A callback function that signals to the runtime that your code has completed. If your function is synchronous,
+     * you must call context.done at the end of execution. If your function is asynchronous, you should not use this
+     * callback.
+     *
+     * @param err A user-defined error to pass back to the runtime. If present, your function execution will fail.
+     * @param result An object containing output binding data.
+     */
     done: IDoneCallback;
+    /**
+     * HTTP request object. Provided to your function when using HTTP Bindings.
+     */
+    req?: IRequest;
+    /**
+     * HTTP response object. Provided to your function when using HTTP Bindings.
+     */
+    res?: {
+        [key: string]: any;
+    };
+}
+/**
+ * HTTP request object. Provided to your function when using HTTP Bindings.
+ */
+export interface IRequest {
+    /**
+     * HTTP request method used to invoke this function.
+     */
+    method: string;
+    /**
+     * Request URL.
+     */
+    url: string;
+    /**
+     * HTTP request headers.
+     */
+    headers: {
+        [key: string]: string;
+    };
+    /**
+     * Query string parameter keys and values from the URL.
+     */
+    query: {
+        [key: string]: string;
+    };
+    /**
+     * Route parameter keys and values.
+     */
+    params: {
+        [key: string]: string;
+    };
+    /**
+     * The HTTP request body
+     */
+    body?: any;
+    /**
+     * The HTTP request body as a UTF-8 string
+     */
+    rawbody?: any;
 }
 export interface IExecutionContext {
+    /**
+     * A unique GUID per function invocation.
+     */
     invocationId: string;
+    /**
+     * The name of the function that is being invoked. The name of your function is always the same as the
+     * name of the corresponding function.json's parent directory.
+     */
     functionName: string;
+    /**
+     * The directory your function is in (this is the parent directory of this function's function.json).
+     */
     functionDirectory: string;
 }
+export interface IBindingDefinition {
+    /**
+     * The name of your binding, as defined in function.json.
+     */
+    name: string;
+    /**
+     * The type of your binding, as defined in function.json.
+     */
+    type: string;
+    /**
+     * The direction of your binding ('in', 'out', or 'inout'), as defined in function.json.
```

Comment:
Do you think we should elaborate on this more? Slash do we do this in the documentation? This has always been fairly confusing.